public class PilaLD implements PilaTDA {
    private static class Nodo {
        int info;
        Nodo sig;
    }

    private Nodo primero;

    public void InicializarPila() {
        primero = null;
    }

    public void Apilar(int x) {
        Nodo nuevo = new Nodo();
        nuevo.info = x;
        nuevo.sig = primero;
        primero = nuevo;
    }

    public void Desapilar() {
        primero = primero.sig;
    }

    public int Tope() {
        return primero.info;
    }

    public boolean PilaVacia() {
        return primero == null;
    }
}
